Inferences/Induction.hpp

```diff
@@ -40,20 +40,23 @@ using namespace Kernel;
 using namespace Saturation;
 
 class TermReplacement : public TermTransformer {
+public:
+  TermReplacement(Term* o, TermList r) : _o(o), _r(r) {} 
+  TermList transformSubterm(TermList trm) override;
+private:
+  Term* _o;
+  TermList _r;
+};
 
+class SkolemSquashingTermReplacement : public TermTransformer {
```

Comment:
What about inheriting from `TermReplacement` instead? That way we can de-duplicate the member fields `_o` and `_r`, and `transformSubterm` is overridden completely anyhow.